pyanaconda/ui/webui/__init__.py

```diff
@@ -141,14 +142,39 @@ def _run_webui(self):
         # FIXME: We probably want to move this to early execution similar to what we have on live
         profile_name = FIREFOX_THEME_DEFAULT
 
-        proc = startProgram(["/usr/libexec/webui-desktop",
-                             "-t", profile_name, "-r", str(int(self.remote)),
-                             "/cockpit/@localhost/anaconda-webui/index.html"],
-                            reset_lang=False)
-        log.debug("cockpit web view has been started")
-        with open(self._viewer_pid_file, "w") as f:
-            f.write(repr(proc.pid))
-        proc.wait()
+        try:
+            proc = startProgram(
+                ["/usr/libexec/webui-desktop",
+                 "-t", profile_name, "-r", str(int(self.remote)),
+                 "/cockpit/@localhost/anaconda-webui/index.html"],
+                reset_lang=False
+            )
+
+            with open(self._viewer_pid_file, "w") as f:
+                f.write(repr(proc.pid))
+
+            (output_string, err_string) = proc.communicate()
+
+            output_string = output_string.decode("utf-8")
+            if output_string and output_string[-1] != "\n":
+                output_string = output_string + "\n"
+
+            log.debug("ddddd output: %s errors: %s", output_string, err_string)
+
+            with program_log_lock:
+                if output_string:
+                    for line in output_string.splitlines():
+                        log.info(line)
+
+                if err_string:
+                    log.error("Errors from webui-desktop:")
+                    for line in err_string.splitlines():
+                        log.error(line)
+            proc.wait()
+        except OSError as e:
+            with program_log_lock:
+                log.error("Error running webui-desktop: %s", e.strerror)
```

Comment:
We're usually just doing 
```python
log.error(".... %s", e)
```
It will automatically get you the message from the Error.